p2p/base_reactor.go

```diff
@@ -41,19 +46,79 @@ type Reactor interface {
 	// Receive is called by the switch when an envelope is received from any connected
 	// peer on any of the channels registered by the reactor
 	Receive(Envelope)
+
+	// QueueUnprocessedEnvelope is called by the switch when an unprocessed
+	// envelope is received. Unprocessed envelopes are immediately buffered in a
+	// queue to avoid blocking. Incoming messages are then passed to a
+	// processing function. The default processing function unmarshals the
+	// messages in the order the sender sent them and then calls Receive on the
+	// reactor. The queue size and the processing function can be changed via
+	// passing options to the base reactor.
+	// 	QueueUnprocessedEnvelope(e UnprocessedEnvelope)
```

Comment:
leaving a comment to link piece of code and PR to an issue. we need to handle this at some point. either remove it or re-add it

Reply:
+1

There is a lot of commented code in here. We could either remove the commented code and create an issue, or uncomment it.

If we decide to remove it, then we should remove all references to the parallel processing including the queue sizes, etc